app/models/form/setup/questions/scheme_id.rb

```diff
@@ -0,0 +1,38 @@
+class Form::Setup::Questions::SchemeId < ::Form::Question
+  def initialize(_id, hsh, page)
+    super("scheme_id", hsh, page)
+    @check_answer_label = "Scheme name"
+    @header = "What scheme is this log for?"
+    @hint_text = "Enter scheme name or postcode"
+    @type = "select"
+    @answer_options = answer_options
+    @derived = true unless FeatureToggle.supported_housing_schemes_enabled?
+  end
+
+  def answer_options
+    answer_opts = {}
+    return answer_opts unless ActiveRecord::Base.connected?
+
+    Scheme.select(:id, :service_name).each_with_object(answer_opts) do |scheme, hsh|
+      hsh[scheme.id.to_s] = scheme.service_name
+      hsh
+    end
+  end
+
+  def displayed_answer_options(case_log)
+    return {} unless case_log.created_by
+
+    user_org_scheme_ids = Scheme.where(organisation_id: case_log.created_by.organisation_id).map(&:id).map(&:to_s)
```

Comment:
I think using `select(:id)` as above avoids needing to loop over the results just to extract that?